torchvision/prototype/features/_image.py

```diff
@@ -17,6 +17,9 @@ class ColorSpace(StrEnum):
     OTHER = 0
     GRAYSCALE = 1
     RGB = 3
+    # On io, we currently support Alpha transparency for Grayscale and RGB. We also support palette for PNG.
+    # Our enum must know about these. We should also allow for users to strip out the alpha transparency which is
+    # currently not supported by any colour transform.
```

Comment:
> We also support palette for PNG.

Could you add a link for that? Otherwise, we are talking about this enum, correct?

https://github.com/pytorch/vision/blob/95d418970e6dbf2e4d928a204c4e620da7bccdc0/torchvision/io/image.py#L16-L31

I'll send a PR to include the `RGBA` and `GRAYSCALE_ALPHA` into the enum and also provide conversion functions for them.

Reply:
Looking into this, conversion from RGBA to RGB is more complicated. For a proper conversion we need to know the background color. We could assume white which will probably work for most cases, but we probably should discuss this first.

Reply:
Palettes are [supported](https://github.com/pytorch/vision/blob/main/torchvision/csrc/io/image/cpu/decode_png.cpp#L107-L109) if the user reads with `IMAGE_READ_MODE_UNCHANGED`. The outcome will be an image with a single channel that has integer IDs for every colour. This is ideal for reading masks that have integer IDs for the classes.

Reply:
Paletted images can be also used to store segmentation masks e.g. VOC